### ISSUE TITLE
Authors Filter seems producing Uncaught InvalidArgumentException

### ISSUE THREAD

**Issue by abdusfauzi** [Automattic/facebook-instant-articles-wp#126]
1. There was no `<author>` in my feed xml

2. I was trying to override my author to our custom authors:

```
function vckt_instant_articles_replace_author( $authors, $post_id) {
    $author_id = get_post_field( 'post_author', $post_id );

    $customAuthor = get_post_meta( $post_id, '_vocket_custom_author', true );

    if ( $customAuthor ) {
        unset( $authors );

        $customAuthorURL = get_post_meta( $post_id, '_vocket_custom_author_url', true );
        $customAuthorDesc = get_post_meta( $post_id, '_vocket_custom_author_description', true );

        $author = new stdClass;
        $author->ID            = $author_id;
        $author->display_name  = $customAuthor;
        $author->user_url      = $customAuthorURL;
        $author->bio           = $customAuthorDesc;

        $authors[] = $author;
    }

    return $authors;
}
add_filter('instant_articles_authors', 'vckt_instant_articles_replace_author', 10, 2);
```

However, adding those codes to `functions.php` produce this error:

```
2016/04/13 11:45:58 [error] 5783#5783: *4030 FastCGI sent in stderr: "PHP message: PHP Fatal error:  Uncaught InvalidArgumentException: Method expects this value
----[
object(stdClass)#325 (4) {
  ["ID"]=>
  string(2) "49"
  ["display_name"]=>
  string(13) "Custom Author"
  ["user_url"]=>
  string(21) "http://www.google.com"
  ["bio"]=>
  string(4) "Hola"
}
]----
 to be one of the types
====[
string(40) "Facebook\InstantArticles\Elements\Author"
]==== in /vagrant/thevocket/content/plugins/fb-instant-articles/vendor/facebook/facebook-instant-articles-sdk-php/src/Facebook/InstantArticles/Validators/Type.php:176
Stack trace:
#0 /vagrant/thevocket/content/plugins/fb-instant-articles/vendor/facebook/facebook-instant-articles-sdk-php/src/Facebook/InstantArticles/Validators/Type.php(68): Facebook\InstantArticles\Validators\Type::throwException(Object(stdClass), 'Facebook\\Instan...')
#1 /vagrant/thevocket/content/plugins/fb-instant-articles/vendor/facebook/facebook-instant-articles-sdk-php/src/Facebook/InstantArticles/Validators/Type.php(39): Facebook\InstantArticles\Validators\" while reading response header from upstream, client: 10.0.0.1, server: thevocket.dev, request: "GET /feed/instant-articles HTTP/1.1", upstream: "fastcgi://unix:/tmp/php-fpm.sock:", host: "thevocket.dev"
```

**Comment by bactisme** [comment]
Exactly same here.

1/ It seems to me that  class-instant-articles-post.php line 401 : $this->_post->author should be $this->_post->post_author
2/ Corrected file produce the same error :dancer:

**Comment by banago** [comment]
Same here. It happens in `wp-admin/post-new.php`. view.

It's impossible to have to up on live site, you cannot post anything.

**Comment by diegoquinteiro** [comment]
Thanks for the feedback.

That should be fixed on master.

Can you try?

I'll be releasing v2.6 soon with the fix.

**Comment by banago** [comment]
@diegoquinteiro it is fixed indeed, thanks a lot.

**Comment by diegoquinteiro** [comment]
Thanks you guys for the feedback!

This was fixed in v2.7 we just released.

Please reopen if you still have issues in the new version.
